dataland-frontend/src/components/pages/CompanyCockpitPage.vue

```diff
@@ -2,288 +2,264 @@
+const FRAMEWORKS_MAIN = MAIN_FRAMEWORKS_IN_ENUM_CLASS_ORDER;
+const showAllFrameworks = ref(false);
+const isUserCompanyOwnerOrUploader = ref(false);
+const isUserKeycloakUploader = ref(false);
+const isAnyCompanyOwnerExisting = ref(false);
+
+const latestDocuments = reactive<Record<string, DocumentMetaInfoResponse[]>>({});
+Object.values(DocumentMetaInfoDocumentCategoryEnum).forEach((category) => {
+  latestDocuments[`latest${category}`] = [];
+});
+const chunkSize = 3;
+const companyRoles = [CompanyRole.MemberAdmin, CompanyRole.Member, CompanyRole.CompanyOwner, CompanyRole.DataUploader];
+const isClaimPanelVisible = computed(() => !isAnyCompanyOwnerExisting.value && isCompanyIdValid(props.companyId));
+const frameworksToDisplay = computed(() => (showAllFrameworks.value ? FRAMEWORKS_ALL : FRAMEWORKS_MAIN));
+
+const assignments = unref(companyRoleAssignmentsRef) ?? [];
+const roles = assignments.filter((r) => r.companyId === props.companyId).map((r) => r.companyRole);
+
+/**
+ * Returns the document data array for a given category.
+ */
+const getDocumentData = (category: keyof typeof DocumentMetaInfoDocumentCategoryEnum): DocumentMetaInfoResponse[] => {
+  return latestDocuments[`latest${category}`] || [];
+};
+
+/**
+ * Fetches the aggregated framework data summary for the current company.
+ * Updates the aggregatedFrameworkDataSummary ref with the response.
+ */
+async function getAggregatedFrameworkDataSummary(): Promise<void> {
+  const api = new ApiClientProvider(assertDefined(getKeycloakPromise)()).backendClients.companyDataController;
+  const response = await api.getAggregatedFrameworkDataSummary(props.companyId);
+  aggregatedFrameworkDataSummary.value = response.data as SummaryByType;
+}
+
+/**
+ * Fetches the latest document meta-information for each document category.
+ * Populates the latestDocuments reactive object with the results.
+ */
+async function getMetaInfoForLatestDocuments(): Promise<void> {
+  const api = new ApiClientProvider(assertDefined(getKeycloakPromise)()).apiClients.documentController;
+  for (const value of Object.values(SearchForDocumentMetaInformationDocumentCategoriesEnum)) {
+    const result = await api.searchForDocumentMetaInformation(props.companyId, new Set([value]), undefined, chunkSize);
+    latestDocuments[`latest${value}`] = result.data;
+  }
+}
+
+/**
+ * Navigates to the document overview page for the current company.
+ */
+function routeToDocuments(): void {
+  void router.push({ path: `/companies/${props.companyId}/documents` });
+}
+
+/**
+ * Determines if the user is allowed to upload for a given framework.
+ */
+function isUserAllowedToUploadForFramework(framework: DataTypeEnum): boolean {
+  return isUserCompanyOwnerOrUploader.value || (isFrameworkPublic(framework) && isUserKeycloakUploader.value);
+}
+
+/**
+ * Sets user rights and roles for the current company.
+ * Updates isAnyCompanyOwnerExisting, isUserCompanyOwnerOrUploader, and isUserKeycloakUploader.
```

Comment:
I missed that earlier, sorry
```suggestion
 * Updates isAnyCompanyOwnerExisting, isUserCompanyOwnerOrUploader, isUserKeycloakUploader and isCompanyMemberOrAdmin
```

Reply:
Done